### PR TITLE
adds both blood level and body temp to analyzers and cryo

### PR DESCRIPTION
**Pull request by QuietlyWhisper**
## About the PR

Mostly a port from MS14

Adds temperature and blood level readout to the health analyzer and cryo pod analyzer. If there's any issues with the code I likely can't fix it myself so feel free to suggest/add code.

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
image

**Changelog**

:cl: Partmedia(MS14) and Whisper
- add: Added temperature readouts to health and cryo analyzers
- add: Added blood level readouts to health and cryo analyzers

### CODE REVIEW

File: Content.Client/HealthAnalyzer/UI/HealthAnalyzerWindow.xaml.cs [hunk 1]
```diff
@@ -36,6 +36,9 @@ public void Populate(HealthAnalyzerScannedUserMessage msg)
 
                 text.Append($"{Loc.GetString("health-analyzer-window-entity-health-text", ("entityName", entityName))}\n");
 
+                text.Append(String.Format("Temperature: {0:F1} °C\n", msg.Temperature - 273f));
+                text.Append(String.Format("Blood Level: {0:F1} %\n", msg.BloodLevel * 100));
```
Comment (tom-leys) [inline Content.Client/HealthAnalyzer/UI/HealthAnalyzerWindow.xaml.cs:40]: 1. Can you not get the TemperatureComponent and BloodstreamComponent here, just like `damageable` was obtained?
2. You need to add more localized strings. look where health-analyzer-window-entity-damage-total-text comes from (file search) and add "Blood Level: " there.

File: Content.Server/Medical/CryoPodSystem.cs [hunk 2]
```diff
@@ -163,10 +164,14 @@ private void OnActivateUIAttempt(EntityUid uid, CryoPodComponent cryoPodComponen
...
     private void OnActivateUI(EntityUid uid, CryoPodComponent cryoPodComponent, AfterActivatableUIOpenEvent args)
     {
+        TryComp<TemperatureComponent>(cryoPodComponent.BodyContainer.ContainedEntity, out var temp);
+        TryComp<BloodstreamComponent>(cryoPodComponent.BodyContainer.ContainedEntity, out var bloodstream);
+
         _userInterfaceSystem.TrySendUiMessage(
             uid,
             HealthAnalyzerUiKey.Key,
-            new HealthAnalyzerScannedUserMessage(cryoPodComponent.BodyContainer.ContainedEntity));
+            new HealthAnalyzerScannedUserMessage(cryoPodComponent.BodyContainer.ContainedEntity,
```
Comment (tom-leys) [inline Content.Server/Medical/CryoPodSystem.cs:173]: The fact that this is the only medical data being passed in here shows you must be doing this the wrong way. I don't think you should be adding any code to this function.

File: Content.Server/Medical/HealthAnalyzerSystem.cs [hunk 3]
```diff
@@ -68,8 +70,13 @@ public void UpdateScannedUser(EntityUid uid, EntityUid user, EntityUid? target,
...
                 return;
 
+            TryComp<TemperatureComponent>(target, out var temp);
+            TryComp<BloodstreamComponent>(target, out var bloodstream);
+
             OpenUserInterface(user, healthAnalyzer);
-            _uiSystem.SendUiMessage(healthAnalyzer.UserInterface, new HealthAnalyzerScannedUserMessage(target));
+
+            _uiSystem.SendUiMessage(healthAnalyzer.UserInterface, new HealthAnalyzerScannedUserMessage(target, temp != null ? temp.CurrentTemperature : 0,
+                bloodstream != null ? bloodstream.BloodSolution.FillFraction : 0 /*bloodstream.BloodSolution.FillFraction * 100*/));
```
Comment (tom-leys) [inline Content.Server/Medical/HealthAnalyzerSystem.cs:79]: You shouldn't be modifying this function either, try getting the data where I suggested above.

File: Content.Shared/MedicalScanner/HealthAnalyzerScannedUserMessage.cs [hunk 4]
```diff
@@ -9,10 +9,14 @@ namespace Content.Shared.MedicalScanner;
 public sealed class HealthAnalyzerScannedUserMessage : BoundUserInterfaceMessage
 {
     public readonly EntityUid? TargetEntity;
+    public float Temperature;
+    public float BloodLevel;
```
Comment (tom-leys) [inline Content.Shared/MedicalScanner/HealthAnalyzerScannedUserMessage.cs:13]: Again, you don't need to change this class.

**Review by tom-leys — COMMENTED** [review]
A great feature idea for a PR. Just need to obtain the data in the right place.

**Comment by QuietlyWhisper** [comment]
Probably, It would just need a check is all

**Pull request merged**